Library/Homebrew/cask/lib/hbc/dsl.rb

```diff
@@ -134,9 +134,8 @@ def language_eval
       end
 
       MacOS.languages.map(&Locale.method(:parse)).each do |locale|
-        key = @language_blocks.keys.detect do |strings|
-          strings.any? { |string| locale.include?(string) }
-        end
+        key = @language_blocks.keys.detect { |strings| strings.any? { |string| locale.eql?(string) } } ||
+              @language_blocks.keys.detect { |strings| strings.any? { |string| locale.include?(string) } }
```

Comment:
Maybe we should add a helper method to `Locale`, so we can add a simple unit test. Something like:

```ruby
key = locale.detect(*@language_blocks.keys)
```

Reply:
Gave it a try, but I don’t yet understand the full workings of `Locale` to be able to do it.

I have no problem with closing this over a better solution, though.

Reply:
Basically just move this logic into a method in `Locale`.

Reply:
That’s what I tried and failed to do. I understand the logic of what I need to do, but the code isn’t coming out right. I can’t return the correct thing and it always goes for the default language.

Reply:
Could you maybe push the code anyways, even if it isn't working?

Reply:
Nothing like a bit of sleep to make us understand what we were doing wrong. Got it to work though I haven’t written the test for the helper method.